chapters/part2/representing-text-as-features.md

```diff
@@ -455,3 +455,108 @@ code below.
 <codeblock source="part2/representing-text-as-features/interacting_with_tensors" setup="part2/representing-text-as-features/setup"></codeblock>
 
 </exercise>
+
+
+<exercise id="10" title="How to upload transformer weights and tokenizers to HuggingFace">
+
+One way AllenNLP is commonly used is for fine-tuning transformer models to specific tasks. We host several of these models on our [demo site](https://demo.allennlp.org/), such as a BERT model applied to the [SQuAD v1.1](https://rajpurkar.github.io/SQuAD-explorer/) question-answer task, and a RoBERTa model applied to the [SNLI](https://nlp.stanford.edu/projects/snli/) textual entailment task.
+
+*You can find the code and configuration files used to train these models in the [AllenNLP Models](https://github.com/allenai/allennlp-models) repository.*
+
+This section will show you how to take a fine-tuned transformer model, like one of these, and upload the weights and/or the tokenizer to [HuggingFace’s model hub](https://huggingface.co/models).
+
+*Note that we are talking about uploading only the transformer part of your model, not including any task-specific heads that you’re using.*
+
+First of all, you’ll need to know how a transformer model and tokenizer is actually integrated into an AllenNLP model.
+
+This is usually done by providing your dataset reader with a `PretrainedTransformerTokenizer` and a matching `PretrainedTransformerIndexer`, and then providing your model with the corresponding `PretrainedTransformerEmbedder`.
+
+If your dataset reader and model are already general enough that they can accept any type of tokenizer / token indexer and token embedder, respectively, then the only thing you need to do in order to utilize a pretrained transformer in your model is tweak your training configuration file.
+
+With the RoBERTa SNLI model, for example, the “dataset_reader” part of the config would look like this:
+
+```
+"dataset_reader": {
+  "type": "snli",
+  "tokenizer": {
+    "type": "pretrained_transformer",
+    "model_name": "roberta-large",
+    "add_special_tokens": false
+  },
+  "token_indexers": {
+    "tokens": {
+      "type": "pretrained_transformer",
+      "model_name": "roberta-large",
+      "max_length": 512
+    }
+  }
+}
+```
+
+While the “model” part of the config would look like this:
+
+```
+"model": {
+  "type": "basic_classifier",
+  "text_field_embedder": {
+    "token_embedders": {
+      "tokens": {
+        "type": "pretrained_transformer",
+        "model_name": "roberta-large",
+        "max_length": 512
+      }
+    }
+  },
+  ...
+}
+```
+
+Once you’ve trained your model, just follow these 3 steps to upload the transformer part of your model to HuggingFace.
```

Comment:
This whole section can probably just be replaced by a pointer to one of the sections above, where we already have this information.